testcontainers/elasticsearch.py

```diff
@@ -10,10 +10,50 @@
 #    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 #    License for the specific language governing permissions and limitations
 #    under the License.
+import logging
+import re
+import urllib
+from typing import Dict
+
 from deprecation import deprecated
+
 from testcontainers.core.container import DockerContainer
 from testcontainers.core.waiting_utils import wait_container_is_ready
-import urllib
+
+_FALLBACK_VERSION = 8
+"""This version is used when no version could be detected from the image name."""
+
+
+def _major_version_from_image_name(image_name: str) -> int:
+    """Returns the major version from a container name like 'elasticsearch:8.1.0'
+    If the major version could not be determined, it will use the most recent
+    one (8 at the time of writing 2022-08-11).
+    """
+    version_string = image_name.split(":")[-1]
+    regex_match = re.compile(r"(\d+)\.\d+\.\d+").match(version_string)
+    if not regex_match:
+        logging.warning("Could not determine major version from image name '%s'. Will use %s",
+                image_name, _FALLBACK_VERSION)
+        return _FALLBACK_VERSION
+    else:
+        return int(regex_match.group(1))
+
+
+def _environment_by_version(version: int) -> Dict[str, str]:
+    """Returns environment variables required for each major version to work."""
+    if version == 6:
```

Comment:
Thoughts on using `version <= 6` here for backwards compatibility?

Reply:
elastic.co supports only three major versions at a time, and version 5 has been eol for 2.5 years.

But I have no strong opinion here, and I understand that there will be people out there who still use version 5. (Although, I doubt that they will use testcontainers)

What I can do is to add a separate case for versions <= 5 . For these versions I would use the same setting as for version 6 and log a warning that these are neither supported by elastic.co nor this project. What do you think?

Reply:
I have tested version `5.6.16`, and that runs fine with the environment for version 6

Reply:
Okay, let's leave it as-is.